desktop-src/direct3dhlsl_current/hlsl-method-gatherCmp.md

```diff
@@ -0,0 +1,283 @@
+| in [`<OffsetType> offset`](#offsettype-offset) | The offset applied to the texture coordinates before sampling. |
+| \[ out [`uint status`](#uint-status) \] | The status of the operation. You can't access the status directly; instead, pass the status to the CheckAccessFullyMapped intrinsic function. CheckAccessFullyMapped returns TRUE if all values from the corresponding Sample, Gather, or Load operation accessed mapped tiles in a tiled resource. If any values were taken from an unmapped tile, CheckAccessFullyMapped returns FALSE. |
+
+Types that depend on texture object:
+
+| TextureObject | [`<CoordType>`](#coordtype-coord) | [`<OffsetType>`](#offsettype-offset) |
+| --- | --- | --- |
+| `Texture2D` | `float2` | `int2` |
+| `Texture2DArray` | `float3` | `int2` |
+| `TextureCube` | `float3` | N/A |
+| `TextureCubeArray` | `float4` | N/A |
+
+### `SamplerState samp`
+
+A Sampler state. This is an object declared in an effect file that contains state assignments.
+
+> My alternate text:
+> SamplerState object that determines how sampling will take place.
+
+Type: `SamplerState`
+
+> TBD: Link to document describing SamplerState object.
+
+### `<CoordType> coord`
+
+The texture coordinates.  The argument type is dependent on the texture-object type.
+
+| TextureObject | `<CoordType>` | x,y,z texel coord | Array Index |
+| --------------- | ------------ | ------------ | ----------- |
+| `Texture2D` | `float2` | xy | - |
+| `Texture2DArray` | `float3` | xy | z |
+| `TextureCube` | `float3` | xyz | - |
+| `TextureCubeArray` | `float4` | xyz | w |
+
+### `float compareValue`
+
+A value to compare each against each sampled value.
+
+### `<OffsetType> offset`
+
+An optional texture coordinate offset, which can be used for any texture-object type; the offset is applied to the coord before sampling. The texture offsets need to be static. The argument type is dependent on the texture-object type. For more info, see Applying texture coordinate offsets.
+
+> My alternate text:
+> An integral offset in texels from the sampling coord in the texture.
+> Each component must be an immediate value in the range of `[-8,7]` known at compile time.
+> Cannot be used with TextureCube or TextureCubeArray objects.
+
+Type: Depends on TextureObject.  Types and component meanings are defined in the following table.
+
+| TextureObject | `<OffsetType>` | Integer offset in texels |
+| --------------- | ------------ | ------------ |
+| `Texture2D` | `int2` | xy |
+| `Texture2DArray` | `int2` | xy |
+
+### `uint status`
+
+The status of the operation. You can't access the status directly; instead, pass the status to the CheckAccessFullyMapped intrinsic function. CheckAccessFullyMapped returns TRUE if all values from the corresponding Sample, Gather, or Load operation accessed mapped tiles in a tiled resource. If any values were taken from an unmapped tile, CheckAccessFullyMapped returns FALSE.
+
+## Remarks
+
+None.
```

Comment:
Nothing describes what Red/Green/Blue/Alpha means in the method name.  Wouldn't this be a suitable place for that?